server/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java

```diff
@@ -128,6 +128,21 @@ public static void configure(final Environment environment, boolean useConsole)
         configureESLogging();
         configure(environment.settings(), environment.configFile(), environment.logsFile(), useConsole);
         initializeStatics();
+        // creates a permanent status logger that can watch for StatusLogger events and forward to a real logger
+        configureStatusLoggerForwarder();
+    }
+
+    private static void configureStatusLoggerForwarder() {
+        // the real logger is lazily retrieved here since logging won't yet be setup during clinit of this class
+        var logger = LogManager.getLogger("StatusLogger");
```

Comment:
Lazily retrieved as in this is not a static class member variable, or lazily retrieved as in there is actually meant to be a function here to get the logger on-demand inside the implementation below?

Reply:
Not a static class member. It only needs to be retrieved once, as if it was a static, once logging has been initialized (before this method is called).